96.md

```diff
@@ -0,0 +1,364 @@
+
+### Image Transformations
+
+`Servers` may respond to some image transformation query parameters and ignore those they don't support by serving
+the original image file without transformations.
+
+#### Resizing
+
+Upon upload, `servers` may create resized image variants, such as thumbnails, respecting the original aspect ratio.
+`Clients` may use the `w` query parameter to request an image version with the desired pixel width.
+`Servers` can then serve the variant with the closest width to the parameter value
+or an image variant generated on the fly.
+
+Example: `https://download.domain/download-path/<sha256-file-hash>.png?w=32`
+
+## Deletion
+
+`Servers` must make available the route `https://deletion.domain/deletion-path/<sha256-file-hash>(.ext)` (route from `https://server.example.domain/.well-known/nostr.json`) with `DELETE` method for file deletion.
+The extension is optional as the file hash is the only needed file identification.
+
+`Clients` should send a `DELETE` request to the server url in the format `https://deletion.domain/deletion-path/<sha256-file-hash>.png`. It must include a NIP-98 `Authorization` header.
+
+The `server` should reject deletes from users other than the original uploader. The `pubkey` encoded on the header value identifies the user.
+
+It should be noted that more than one user may have uploaded the same file (so with the same hash). In this case, a delete must not really delete the file but just remove the user's `pubkey` from the file owners list.
+
+The response is similar to the json sent on uploads, but without the `nip96.download_route` field.
+
+## Monetization
+
+This is an **optional** step that may be configured by `servers` at any moment.
+It is a suggested way of monetizing the file storage.
+
+It is a freemium business model, meaning that the file storage is free,
+but it is expected that some users consuming the stored files may "zap" (send)
+a variable amount of cryptocurrency to the server owner wallet.
+
+Although file storage is free of charge, of course the `server` can delete files at any moment at its discrition.
+
+For example, the file storage `server` may fetch [zap receipts](57.md#appendix-e-zap-receipt-event) from it's zap provider pubkey and
+extract embeded media from referenced notes. Then stored files that hasn't received zaps
+for a while may be deleted.
+
+### Server Configuration
+
+`Server` owners should configure their nostr pubkey at the `/.well-known/nostr.json?name=_` path as per [NIP-05](05.md#showing-just-the-domain-as-an-identifier), including `relay` information.
+
+`Server` owners then should add to their nostr account metadata event (kind 0) a lud06 or lud16 field (and/or any future supported cryptocurrency address) pointing to a `zap-enabled` wallet address they own.
+
+It is required to send the event to all relays included on previous NIP-05 configuration. We suggest re-sending the same event to said relays from time to time to make sure they don't get deleted.
```

Comment:
This seems iffy to me, maybe adding support for lightning addresses on the nostr.json endpoint would be preferable?

Reply:
Totally agree! Http file storage servers shouldn't need websockets, just `/.well-known/nostr.json` config. But this [PR](https://github.com/nostr-protocol/nips/pull/552/files) tells that the zap tag can only have pubkey as value. It needs a corresponding kind 0 event to get the updated lud06/16 address.

If @vitorpamplona agrees to update the PR to make the zap tag also accept a NIP-05 address, the server could use just the `https://my.file.server/.well-known/nostr.json` to config the current lud06/16 address.

Like this:
```js
    ["zap", "82341f882b6eabcd2ba7f1ef90aad961cf074af15b9ef44a09f9d2a8fbfbe6a2", "wss://nos.lol", "19"],
    ["zap", "_@my.file.server", "", "1"]
```

Reply:
@vitorpamplona L@_@K

Reply:
It was tried and rejected [here](https://github.com/nostr-protocol/nips/pull/696)